test/run.sh

```diff
@@ -47,14 +49,20 @@ if [ $# -eq 0 ]; then
 	exit 1
 fi
 
-# load config lists
-# contains:
-#   globalTests
-#   testAlias
-#   imageTests
-#   globalExcludeTests
+# declare configuration variables
+declare -a globalTests=()
+declare -A testAlias=()
+declare -A imageTests=()
+declare -A globalExcludeTests=()
+
+# load the default config
 . "$dir/config.sh"
```

Comment:
I'm fine with it either way. In my particular case, the initial goal was to use the stuff from the default config for a derived image, so I coded it this way.

I think the bigger issue is how/when to determine the relative path to individual test cases. I think that maintaining the simple naming convention for test cases currently used in `$imageTests` makes some sense, but it gets complicated if I've replaced the default `config.sh` with my own config. Let's say I wanted to do that and have it run `postgres-basics`, `postgres-initdb`, and `postgis-basics`. One might imagine a custom config that looks like this:

```bash
imageTests[appropriate/postgis] = '
	postgres-basics
	postgres-initdb
	postgis-basics
'
```

Given that configuration, how would the test runner know to look for the first two in `tests/postgres-*` relative to `~/official-images/test` while looking for my added test relative to the directory that the custom config is defined in?

My first thought about this was to have `run.sh` reinitialize `imageTests` to an empty associative array before loading each config, then copy it into a data structure that tracks each iteration of `imageTests` in another associative array keyed by the config file's location, but that seems kind of brittle. It seems like `~/official-images/test/tests/$NAME/` should always be checked if the test directory is not found in a `tests/$NAME` directory that's a sibling of the custom config itself.